src/env.h

```diff
@@ -578,6 +581,14 @@ class Environment {
 
   char* http_parser_buffer_;
 
+  struct AtExitCallback {
+    AtExitCallback* next_;
+    void (*cb_)(void* arg);
+    void* arg_;
+  };
```

Comment:
Just a suggestion, but you could make `AtExitCallback` a forward declaration in the `node` namespace here and only define it right before `AtExit`/`RunAtExit`. That would make sense insofar as the exact layout isn’t part of the API here, and having the definition of the `struct` close to its actual usage might be a tad more readable.